src/formatDate.js

```diff
@@ -50,7 +50,57 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateSplit = date.split(fromFormat[fromFormat.length - 1]);
+  const dateFinal = [];
+  let year = null;
+  let month = null;
+  let day = null;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateSplit[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateSplit[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateSplit[i];
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'YYYY' && year.length === 4) {
+      dateFinal.push(year);
+    }
+
+    if (toFormat[i] === 'YYYY' && year.length === 2) {
+      if (year < 30) {
+        dateFinal.push(`20${year}`);
+      } else {
+        dateFinal.push(`19${year}`);
+      }
+    }
+
+    if (toFormat[i] === 'YY' && year.length === 2) {
+      dateFinal.push(year);
+    }
+
+    if (toFormat[i] === 'YY' && year.length === 4) {
+      dateFinal.push(year.slice(2));
+    }
+
+    if (toFormat[i] === 'MM') {
+      dateFinal.push(month);
+    }
+
+    if (toFormat[i] === 'DD') {
+      dateFinal.push(day);
+    }
```

Comment:
If you'll use object instead variables you'll be able to get rid of this two `if`s